lib/HALTarget/IMU.h

```diff
@@ -0,0 +1,186 @@
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include "IIMU.h"
+#include "Zumo32U4IMU.h"
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and Classes
+ *****************************************************************************/
+
+/** The IMU adapter. */
+class IMU : public IIMU
+{
+public:
+    /**
+     * Constructs the IMU adapter.
+     *
+     */
+    IMU() :
+        IIMU(),
+        m_rawAccelerometerOffsetX(0),
+        m_rawAccelerometerOffsetY(0),
+        m_rawAccelerometerOffsetZ(0),
+        m_rawGyroOffsetX(0),
+        m_rawGyroOffsetY(0),
+        m_rawGyroOffsetZ(0)
+    {
+    }
+
+    /**
+     * Destroys the IMU adapter.
+     */
+    ~IMU()
+    {
+    }
+
+    /**
+     * Initializes the inertial sensors and detects their type.
+     *
+     * @return True if the sensor type was detected succesfully; false otherwise.
+     */
+    bool init();
+
+    /**
+     * Enables all of the inertial sensors with a default configuration.
+     */
+    void enableDefault();
+
+    /**
+     * 	Configures the sensors with settings optimized for turn sensing.
+     */
+    void configureForTurnSensing();
+
+    /**
+     * Takes a reading from the accelerometer and makes the measurements available in a.
+     */
+    void readAccelerometer();
+
+    /**
+     * Takes a reading from the gyro and makes the measurements available in g.
+     */
+    void readGyro();
+
+    /**
+     * Takes a reading from the magnetometer and makes the measurements available in m.
+     */
+    void readMagnetometer();
+
+    /**
+     * Indicates whether the accelerometer has new measurement data ready.
+     *
+     * @return True if there is new accelerometer data available; false otherwise.
+     */
+    bool accelerometerDataReady();
+
+    /**
+     * Indicates whether the gyro  has new measurement data ready.
+     *
+     * @return True if there is new gyro data available; false otherwise.
+     */
+    bool gyroDataReady();
+
+    /**
+     * Indicates whether the magnetometer  has new measurement data ready.
+     *
+     * @return True if there is new magnetometer data available; false otherwise.
+     */
+    bool magnetometerDataReady();
+
+    /**
+     * Get last raw Accelerometer values as a IMUData struct containing values in x, y and z.
+     *
+     * @param[in] accelerationValues  Pointer to IMUData struct.
+     */
+    const void getAccelerationValues(IMUData* accelerationValues);
+
+    /**
+     * Get last raw Gyroscope values as a IMUData struct containing values in x, y and z.
+     *
+     * @param[in] turnRates  Pointer to IMUData struct.
```

Comment:
Unit?

Reply:
Added an explanation why the value is unitless.